src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,49 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_LENGTH = 10;
+    private static final int START_INDEX = 0;
+    private static final int NO_MATCH_INDEX = -1;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ARRAY_LENGTH];
+        values = (V[]) new Object[MAX_ARRAY_LENGTH];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getIndexForPutting(key);
+        if (index == NO_MATCH_INDEX) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        } else {
+            values[index] = value;
```

Comment:
Of course I'm not sure if my variant is good or not, but I just wanted to show it :D
```suggestion
         int index = getIndexForPutting(key);
        if (index == NO_MATCH_INDEX) {
            keys[size] = key;
            index = size;           
            size++;
        } 
        values[index] = value;
```

Reply:
I'm not seen your entire code, but with my solution, if ain't "break" from the method, it's not work